### PR TITLE
CALS-5683 Changing minor child field

### PR DESCRIPTION
**Pull request by nasar-cambria**
#### Which User story does this relate to (link)?
- https://osi-cwds.atlassian.net/browse/CALS-5683

#### Brief feature description
- To change dropdown fields to a radio buttons.

#### Does this PR block/effect other stories?
- NA

#### Do any commands needs to be run in order to run the project with this PR?
- NA

#### Code coverage percent diff as a result of this PR?
- NA

### CODE REVIEW

**Comment by nasar-cambria** [comment]
jenkins re-test this please

**Comment by anuroop15** [comment]
@nasar-cambria: Unable to change the radio button, when you add another minor child. Please fix this.

File: app/javascript/rfa_forms/rfa01a_edit_view/minorCardField.js [hunk 1]
```diff
@@ -91,20 +92,20 @@ export class MinorCardField extends React.Component {
...
+          onFieldChange={(event) => this.props.onFieldChange(this.props.index, event.target.value, 'child_financially_supported')} />
+        <YesNoRadioComponent
+          idPrefix='child_adopted'
           value={minor.child_adopted}
           selectClassName={'reusable-select'}
           optionList={yesNo.items}
           label={isRelationShipToApplicantObject ? 'Is this child adopted? (required)' : 'Is this child adopted?'}
-          onChange={(event) => this.props.onFieldChange(this.props.index, dictionaryNilSelectValue(event.target.options), 'child_adopted')} />
+          onFieldChange={(event) => this.props.onFieldChange(this.props.index, event.target.value, 'child_adopted')} />
       </form>
```
Comment (ratneshraval) [inline app/javascript/rfa_forms/rfa01a_edit_view/minorCardField.js]: Can you remove the `<form>` tags in this file?

File: app/javascript/rfa_forms/rfa01a_edit_view/minorCardField.js [hunk 2]
```diff
@@ -91,20 +92,20 @@ export class MinorCardField extends React.Component {
           value={getDictionaryId(minor.gender)}
           label={isRelationShipToApplicantObject ? 'Gender (required)' : 'Gender'}
           onChange={(event) => this.props.onFieldChange(this.props.index, dictionaryNilSelect(event.target.options), 'gender')} />
-        <DropDownField gridClassName='col-md-4'
-          id='child_financially_supported'
+        <YesNoRadioComponent
+          label={isRelationShipToApplicantObject ? 'Do you financially support this child? (required)' : 'Do you financially support this child?'}
```
Comment (ratneshraval) [inline app/javascript/rfa_forms/rfa01a_edit_view/minorCardField.js]: so there are many conditionals in this file, which will affect your branch coverage for testing.
Create a label at top of render based on the `isRelationShipToApplicantObject` and then append that throughout, like this
```
const isRequiredLabel = this.isRelationShipToApplicantObject() ? ' (required)' : ''
...
label={'Do you financially support this child?' + isRequiredLabel}
```

File: spec/acceptance/rfa_forms/rfa_01a_application_spec.rb [hunk 3]
```diff
@@ -8,10 +8,10 @@
 
   before(:each) do 
     visit root_path
-	end
-	scenario 'Dashboard page', set_auth_header: true do
-	    expect(page).to have_button('Create RFA Application (Form 01)')
-  	end
+    end
+    scenario 'Dashboard page', set_auth_header: true do
```
Comment (ratneshraval) [inline spec/acceptance/rfa_forms/rfa_01a_application_spec.rb]: looks like there is still alignment issue here.

File: spec/acceptance/rfa_forms/rfa_01a_application_spec.rb [hunk 4]
```diff
@@ -63,8 +63,8 @@
   end
 
   scenario 'validate remove button, applicant card', set_auth_header: true do
-  	click_button 'Create RFA Application (Form 01)'
-  	expect(page).to have_content 'Rfa-01A Section Summary'
+    click_button 'Create RFA Application (Form 01)'
```
Comment (ratneshraval) [inline spec/acceptance/rfa_forms/rfa_01a_application_spec.rb]: 👍 on not doing `visit root_path` again.

File: app/javascript/rfa_forms/rfa01a_edit_view/minorCardField.js [hunk 5]
```diff
@@ -60,10 +61,12 @@ export class MinorCardField extends React.Component {
   }
 
   render () {
+    const isRequiredLabel = this.isRelationShipToApplicantObject() ? ' (required)' : ''
+    const child = this.props.index
```
Comment (Nasir2712) [inline app/javascript/rfa_forms/rfa01a_edit_view/minorCardField.js]: can you rename child to index

File: app/javascript/rfa_forms/rfa01a_edit_view/minorCardField.js [hunk 6]
```diff
@@ -89,23 +92,23 @@ export class MinorCardField extends React.Component {
...
+          label={'Do you financially support this child?' + isRequiredLabel}
+          idPrefix={'child_financially_supported' + index}
           value={minor.child_financially_supported}
           selectClassName={'reusable-select'}
           optionList={yesNo.items}
-          label={isRelationShipToApplicantObject ? 'Do you financially support this child? (required)' : 'Do you financially support this child?'}
-          onChange={(event) => this.props.onFieldChange(this.props.index, dictionaryNilSelectValue(event.target.options), 'child_financially_supported')} />
-        <DropDownField gridClassName='col-md-4'
-          id='child_adopted'
+          onFieldChange={(event) => this.props.onFieldChange(this.props.index, event.target.value, 'child_financially_supported')} />
```
Comment (ratneshraval) [inline app/javascript/rfa_forms/rfa01a_edit_view/minorCardField.js]: You have `const index` declared up on line 65 but still using `this.props.index` here and other places, stick with one approach.

Comment (nasar-cambria) [reply app/javascript/rfa_forms/rfa01a_edit_view/minorCardField.js]: good catch, I'll fix that

File: app/javascript/rfa_forms/rfa01a_edit_view/minorCardField.js [hunk 7]
```diff
@@ -60,10 +61,12 @@ export class MinorCardField extends React.Component {
   }
 
   render () {
+    const isRequiredLabel = this.isRelationShipToApplicantObject() ? ' (required)' : ''
+    const index = this.props.index
     const minor = this.props.minorChild
     const isRelationShipToApplicantObject = this.isRelationShipToApplicantObject()
```
Comment (ratneshraval) [inline app/javascript/rfa_forms/rfa01a_edit_view/minorCardField.js]: Looks like you aren't using this constant anymore.

File: spec/acceptance/rfa_forms/rfa_01a_application_spec.rb [hunk 8]
```diff
@@ -6,12 +6,12 @@
 
 RSpec.feature 'RFA01A', js: true do
 
-  before(:each) do 
-    visit root_path
-	end
-	scenario 'Dashboard page', set_auth_header: true do
-	    expect(page).to have_button('Create RFA Application (Form 01)')
-  	end
+before(:each) do
```
Comment (ratneshraval) [inline spec/acceptance/rfa_forms/rfa_01a_application_spec.rb]: Still not aligned properly. You should install ruby formatter for your editor to auto format files you save.

**Comment by anuroop15** [comment]
Please fix the console warning:
 Invalid DOM property `class`. Did you mean `className`?
    in div (created by MinorCardField)

File: test/javascript/rfa_forms/rfa01a_edit_view/minorChildrenCard_tests.js [hunk 9]
```diff
@@ -68,14 +68,31 @@ describe('Verify MinorCardFields', function () {
...
-  it('verify child_financially_supported field', () => {
-    let relationShipField = minorChildCardComp.find('#child_financially_supported')
-    relationShipField.simulate('change', selectedYes)
-    expect(onFieldChangeSpy).toHaveBeenCalledWith(0, '2', 'child_financially_supported')
-  })
-  it('verify child adopted field', () => {
-    let relationShipField = minorChildCardComp.find('#child_adopted')
-    relationShipField.simulate('change', selectedYes)
-    expect(onFieldChangeSpy).toHaveBeenCalledWith(0, '2', 'child_adopted')
+  describe('#YesNoRadioComponent', () => {
```
[the remark this answers is not in the document]
Comment (nasar-cambria) [reply test/javascript/rfa_forms/rfa01a_edit_view/minorChildrenCard_tests.js]: @anuroop15  this story is to just change select drop down to radio button. I am not adding or removing any new fields here.

**Comment by nasar-cambria** [comment]
jenkins re-test this please

**Comment by nasar-cambria** [comment]
jenkins re-test this please

**Pull request merged**